service/src/main/java/bio/terra/landingzone/service/iam/SamService.java

```diff
@@ -0,0 +1,182 @@
+package bio.terra.landingzone.service.iam;
+
+import bio.terra.common.exception.ForbiddenException;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.common.sam.SamRetry;
+import bio.terra.common.sam.exception.SamExceptionFactory;
+import bio.terra.common.tracing.OkHttpClientTracingInterceptor;
+import bio.terra.landingzone.library.configuration.SamConfiguration;
+import io.opencensus.contrib.spring.aop.Traced;
+import io.opencensus.trace.Tracing;
+import java.util.List;
+import java.util.UUID;
+import okhttp3.OkHttpClient;
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.api.ResourcesApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi;
+import org.broadinstitute.dsde.workbench.client.sam.model.CreateResourceRequestV2;
+import org.broadinstitute.dsde.workbench.client.sam.model.FullyQualifiedResourceId;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SamService {
+  private static final Logger logger = LoggerFactory.getLogger(SamService.class);
+  private final SamConfiguration samConfig;
+  private final OkHttpClient commonHttpClient;
+
+  @Autowired
+  public SamService(SamConfiguration samConfig) {
+    this.samConfig = samConfig;
+    this.commonHttpClient =
+        new ApiClient()
+            .getHttpClient()
+            .newBuilder()
+            .addInterceptor(new OkHttpClientTracingInterceptor(Tracing.getTracer()))
+            .build();
+  }
+
+  private ApiClient getApiClient(String accessToken) {
+    // OkHttpClient objects manage their own thread pools, so it's much more performant to share one
```

Comment:
Copied this approach of sharing a common `OkHttpClient` from WSM: https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/iam/SamService.java